config/routes.rb

```diff
@@ -21,6 +21,8 @@
 
   api :v1, default: true do
     resources :exercises do
+      post :save_image, on: :member
```

Comment:
Using this `resources :attachments` approach would also let us easily add an `:index` action, which would be good because for security reasons I'll likely be removing the list of attachments from the Exercise JSON (so that we don't accidentally show a student an image from the author's solution) -- the `:index` action can check that the requestor is author-like, and this would eventually let us have a gallery of uploaded images to show authors.  There would still be a security concern with people making derived copies of exercises and then viewing the attachments, but I think we could get around that by only transferring to the derived copy those attachments that are publicly visible on the exercise or in the solution.

Also, if you want to keep this code out of the `ExercisesController`, you can make an `AttachmentsController` which could also eventually be used when we do upload attachments to independent solutions.